Make sure you can use the bundle without having twig installed

Nyholm opened the pull request:
Currently the bundle actually has a hard dependency on Twig. 
With this PR you may opt-out of using Twig.

Nyholm commented:
Hm, Im not sure if this really works. I still get: 

> Symfony\Component\Debug\Exception\ClassNotFoundException: Uncaught exception 'Symfony\Component\Debug\Exception\ClassNotFoundException' with message 'Attempted to load class "Twig_Extension" from the global namespace.Did you forget a "use" statement?' in /home/sites/foo/bar/vendor/ekino/newrelic-bundle/Twig/NewRelicExtension.php:20

jderusse commented on DependencyInjection/EkinoNewRelicExtension.php:
```diff
@@ -108,5 +108,9 @@ public function load(array $configs, ContainerBuilder $container)
             ->replaceArgument(5, $config['using_symfony_cache'])
         ;
+
+        if (!$config['twig_enabled']) {
```
What's about checking if the class `Twig_Environment` exists?

Tobion replied:
Agree, the flex way is to enable things when they are installed.

Nyholm replied:
Sure, will do

Nyholm commented:
Finally I found this PR again and completed it. 

I also noticed that @ruudk did #128 which is a duplicate of this PR.

Tobion commented on DependencyInjection/Configuration.php:
```diff
@@ -29,6 +29,7 @@ public function getConfigTreeBuilder()
             ->fixXmlConfig('deployment_name')
             ->children()
                 ->booleanNode('enabled')->defaultTrue()->end()
+                ->booleanNode('twig')->defaultTrue()->end()
```
just set the default to `class_exists(\Twig_Environment::class)` or maybe we on't need a config at all

Nyholm replied:
I may want to turn this off even though I have Twig installed.

ruudk replied:
So something like this:
```php
->booleanNode('twig')->defaultValue(class_exists(\Twig_Environment::class))->end()
```

Nyholm commented:
@Tobion Are you okey with the current state of this PR?

Tobion commented:
As I proposed in https://github.com/ekino/EkinoNewRelicBundle/pull/126/files#r173443429 the default should be changed.

Nyholm commented:
I've updated the PR and rebased on master. 

I just got write permissions to this repo but I do not want to merge my own PRs without approval. Could you do a final review?

Nyholm commented:
Thank you
